### PR TITLE
[posix-trel] retry adding interface address if failed

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit enhances `TREL UDP6` to retry adding the link-local interface address until it's successful.

### CODE REVIEW

File: src/posix/platform/trel_udp6.cpp [hunk 1]
```diff
@@ -405,6 +413,102 @@ static void ReceivePacket(int aSocket, otInstance *aInstance)
...
+    {
+        if (rta->rta_type != IFA_ADDRESS)
+        {
+            continue;
+        }
+
+        {
+            otIp6Address addr;
+            memcpy(&addr, RTA_DATA(rta), sizeof(addr));
+            bool isInterfaceAddress = memcmp(&sInterfaceAddress, RTA_DATA(rta), sizeof(sInterfaceAddress)) == 0;
```
Comment (abtink) [inline src/posix/platform/trel_udp6.cpp:437]: Move bool variable declaration before `memcpy()` and next to `addr`? (so all variables are at the top of scope)

File: src/posix/platform/trel_udp6.cpp [hunk 2]
```diff
@@ -405,6 +413,102 @@ static void ReceivePacket(int aSocket, otInstance *aInstance)
...
+            }
+
+            if (aNetlinkMessage->nlmsg_type == RTM_NEWADDR)
+            {
+                otLogInfoPlat("[trel] Interface address added successfully.");
+                sAddInterfaceAddressDeadline = 0;
+            }
+            else if (aNetlinkMessage->nlmsg_type == RTM_DELADDR)
+            {
+                otLogWarnPlat("[trel] Interface address removed unexpectedly.");
```
Comment (abtink) [inline src/posix/platform/trel_udp6.cpp:451]: How about moving this log line under `if (sAddInterfaceAddressDeadline == 0)` block?

If I recall correctly, on some/most linux platforms, whenever we request an address add, we can see a `RTM_DELADDR` first then followed by a `RTM_NEWADDR` (independent of whether the address was present on netif before). In such a case, we will see this warning log all the time (even when the address is being added ok).

File: src/posix/platform/trel_udp6.cpp [hunk 3]
```diff
@@ -198,6 +197,15 @@ static void RemoveUnicastAddress(const otIp6Address *aUnicastAddress)
     UpdateUnicastAddress(aUnicastAddress, /* aToAdd */ false);
 }
 
+static void HandleAddInterfaceAddressTimeout(void)
+{
+    otLogWarnPlat("[trel] Failed to add TREL interface address, will retry after %d seconds.",
```
Comment (abtink) [inline src/posix/platform/trel_udp6.cpp:202]: I think when sent this log, it's already timed out and we are going to try again immediately. How about something like this?
```suggestion
    otLogWarnPlat("[trel] Failed to add TREL interface address after %d seconds. Trying to add again",
```

File: src/posix/platform/trel_udp6.cpp [hunk 4]
```diff
@@ -751,6 +900,23 @@ void platformTrelProcess(otInstance *aInstance, const fd_set *aReadFdSet, const
...
+
+    if (sAddInterfaceAddressDeadline > 0)
+    {
+        uint64_t now = otPlatTimeGet();
+
+        if (sAddInterfaceAddressDeadline <= now)
+        {
+            sAddInterfaceAddressDeadline = 0;
+
+            HandleAddInterfaceAddressTimeout();
```
Comment (abtink) [inline src/posix/platform/trel_udp6.cpp:916]: Do we need to move this and the checks above under `#if OPENTHREAD_CONFIG_POSIX_TREL_USE_NETLINK_SOCKET`?

I think the `HandleAddInterfaceAddressTimeout()` is now only defined under this config above, right?

**Review by abtink — APPROVED** [review]
Looks great. Thanks @simonlingoogle 👍 

Some smaller questions/suggestions below.

File: src/posix/platform/trel_udp6.cpp (hunk 2 above)
Comment (simonlingoogle) [reply src/posix/platform/trel_udp6.cpp:451]: Make sense to me.

File: src/posix/platform/trel_udp6.cpp (hunk 4 above)
Comment (simonlingoogle) [reply src/posix/platform/trel_udp6.cpp:916]: I will make sure everything works in both configurations.

**Review by jwhui — APPROVED** [review]

**Pull request merged**